nodes/referral_inferencer.py

```diff
@@ -0,0 +1,84 @@
+import json
+from langchain_core.messages import HumanMessage, SystemMessage
+
+from models.context import Context
+from models.llm_result import ClassifierResult, ActionsSummaryResult, ReferralPossibilityResult
+from utils.llm import invoke_llm
+
+
+def infer_referral_possibility(
+    context: Context,
+    classified_category: ClassifierResult,
+    actions_summary: ActionsSummaryResult,
+    dry_run: bool = False,
+) -> ReferralPossibilityResult:
+    system_prompt = f"""\
+You are a referral possibility analyzer that evaluates whether it's still possible to get a referral after the user has taken specific actions.
+
+You will be given:
+1. The conversation history
+2. The classified conversation category
+3. A summary of actions that were taken
+
+Your task is to analyze whether a referral is still achievable and provide:
+1. A clear yes/no answer on referral possibility
+2. Your confidence level in this assessment
+3. Reasoning based on the conversation context and actions taken
+4. Next steps that could help secure the referral (if possible)
```

Comment:
ditto, why do we need this attribute? Will the topics suggester handle the next-step-like problem?